### PR TITLE
Fix flaky bundle tests.

### PR DESCRIPTION
**Pull request by wu-hui**
They are flaky because Android SDK does not guarantee all progress callbacks are *complete* before the task resolves.

Whether or not it should guarantee this is another topic.

### CODE REVIEW

**Comment by jonsimantov** [comment]
> They are flaky because Android SDK does not guarantee all progress callbacks are _complete_ before the task resolves.
> 
> Whether or not it should guarantee this is another topic.

Is this the issue that will be fixed in the next Android release? If so, we should revert this PR after that fix goes in (and we update to that dependency).

File: firestore/integration_test_internal/src/bundle_test.cc [hunk 1]
```diff
@@ -65,6 +65,15 @@ std::string CreateTestBundle(Firestore* db) {
   return CreateBundle(db->app()->options().project_id());
 }
 
+void SetPromiseValueWhenUpdateIsFinal(
+    LoadBundleTaskProgress progress,
+    std::promise<void>& final_update_received) {
```
[the remark this answers is not in the document]
Comment (var-const) [reply firestore/integration_test_internal/src/bundle_test.cc:70]: Nit: please include `<future>` (you might need to add a `// NOLINT(build/c++11)` comment to it).

**Review by var-const — APPROVED** [review]

File: firestore/integration_test_internal/src/bundle_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (wu-hui) [reply firestore/integration_test_internal/src/bundle_test.cc:70]: It is already included, it is already used by one other test.

**Comment by wu-hui** [comment]
> > They are flaky because Android SDK does not guarantee all progress callbacks are _complete_ before the task resolves.
> > Whether or not it should guarantee this is another topic.
> 
> Is this the issue that will be fixed in the next Android release? If so, we should revert this PR after that fix goes in (and we update to that dependency).

We have not decided whether or not we will change Android yet..there are arguments against it. Will revert this PR if that is what we end up doing.

**Pull request merged**